eclipsesource/theia
# Performance measurement

#### What it does


This commit adds a script for performance measurement using the Chrome DevTools tracing.
Currently the startup time of the browser-app is being measured.
The script uses Puppeteer so other processes can be measured in the future as well.
For more information see the `README.md` in the scripts/performance directory.

This PR is intended as the first step to an eventual performance test suite which could also be integrated into the CI tooling.
See https://github.com/eclipse-theia/(remove this parenthesis when opening against Theia)theia/issues/9722 for more information about this topic.

Example output:
```
[Performance][Startup][1] Largest Contentful Paint (LCP): 2.95 seconds
[Performance][Startup][2] Largest Contentful Paint (LCP): 2.823 seconds
[Performance][Startup][3] Largest Contentful Paint (LCP): 2.829 seconds
[Performance][Startup][4] Largest Contentful Paint (LCP): 2.835 seconds
[Performance][Startup][5] Largest Contentful Paint (LCP): 2.778 seconds
[Performance][Startup][6] Largest Contentful Paint (LCP): 2.795 seconds
[Performance][Startup][7] Largest Contentful Paint (LCP): 2.772 seconds
[Performance][Startup][8] Largest Contentful Paint (LCP): 2.672 seconds
[Performance][Startup][9] Largest Contentful Paint (LCP): 2.706 seconds
[Performance][Startup][10] Largest Contentful Paint (LCP): 2.804 seconds
[Performance][Startup][MEAN] Largest Contentful Paint (LCP): 2.796 seconds
[Performance][Startup][STDEV] Largest Contentful Paint (LCP): 0.072 seconds
```

Contributed on behalf of STMicroelectronics

Signed-off-by: sgraband@eclipsesource.com
#### How to test

See the README on how you can run the script manually or run the `performance:startup` node script in the root directory to measure the startup time. You can also adjust the parameters to for example execute more iterations or to run the test headful.
Additionally, you can import the generated files in `scripts/performance/profiles` in the performance tab of the Chrome DevTools.
#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
